pkg/logger/otelzap/otelzap.go

```diff
@@ -0,0 +1,249 @@
+package otelzap
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	otellog "go.opentelemetry.io/otel/log"
+	otelglobal "go.opentelemetry.io/otel/log/global"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	oteltrace "go.opentelemetry.io/otel/trace"
+
+	"go.uber.org/zap/zapcore"
+)
+
+// OtelZapCore is a zapcore.Core implementation that exports logs to OpenTelemetry
+// It implements the zapcore.Core interface and uses OpenTelemetry's logging API
+type OtelZapCore struct {
+	logger       otellog.Logger
+	fields       []zapcore.Field
+	levelEnabler zapcore.LevelEnabler
+}
+
+type Option func(c *OtelZapCore)
+
+// NOTE: [beholder.SetGlobalOtelProviders](https://github.com/smartcontractkit/chainlink-common/blob/27faefc9ce454c8aa2b1b7484e377ea3e8996bba/pkg/beholder/global.go#L50)
+// must be called before using `NewOtelLogger` otherwise otelglobal.GetLoggerProvider() returns noop provider
+func NewOtelLogger(name string) otellog.Logger {
+	return otelglobal.GetLoggerProvider().Logger(name)
+}
+
+// NewOtelCore initializes an OpenTelemetry Core for exporting logs in OTLP format
+func NewOtelZapCore(logger otellog.Logger, opts ...Option) zapcore.Core {
```

Comment:
No need to repeat `OtelZap` since we are already in `package otelzap`:
```suggestion
func NewCore(logger otellog.Logger, opts ...Option) zapcore.Core {
```